src/main/java/io/strimzi/kafka/bridge/http/HttpBridge.java

```diff
@@ -206,29 +206,37 @@ private void createConsumer(RoutingContext routingContext) {
         this.httpBridgeContext.setOpenApiOperation(HttpOpenApiOperations.CREATE_CONSUMER);
 
         JsonObject body = routingContext.getBodyAsJson();
-        EmbeddedFormat format = EmbeddedFormat.from(body.getString("format", "binary"));
+        SinkBridgeEndpoint<byte[], byte[]> sink = null;
 
-        final SinkBridgeEndpoint sink = new HttpSinkBridgeEndpoint<>(this.vertx, this.bridgeConfig, this.httpBridgeContext,
-                format, new ByteArrayDeserializer(), new ByteArrayDeserializer());
+        try {
+            EmbeddedFormat format = EmbeddedFormat.from(body.getString("format", "binary"));
 
-        sink.closeHandler(s -> {
-            httpBridgeContext.getHttpSinkEndpoints().remove(sink.name());
-        });
+            sink = new HttpSinkBridgeEndpoint<>(this.vertx, this.bridgeConfig, this.httpBridgeContext,
+                    format, new ByteArrayDeserializer(), new ByteArrayDeserializer());
 
-        try {
+            sink.closeHandler(endpoint -> {
+                httpBridgeContext.getHttpSinkEndpoints().remove(endpoint.name());
+            });        
             sink.open();
 
-            sink.handle(new HttpEndpoint(routingContext), consumerId -> {
-                httpBridgeContext.getHttpSinkEndpoints().put(consumerId.toString(), sink);
-                timestampMap.put(consumerId.toString(), System.currentTimeMillis());
+            sink.handle(new HttpEndpoint(routingContext), s -> {
+                SinkBridgeEndpoint<byte[], byte[]> endpoint = (SinkBridgeEndpoint<byte[], byte[]>) s;
+                httpBridgeContext.getHttpSinkEndpoints().put(endpoint.name(), endpoint);
+                timestampMap.put(endpoint.name(), System.currentTimeMillis());
             });
         } catch (Exception ex) {
-            sink.close();
+            if (sink != null) {
+                sink.close();
+            }
+            HttpResponseStatus responseStatus = (ex instanceof IllegalArgumentException) ? 
```

Comment:
Sorry but I don't understand your concern about "could propagate out of the try block via other method calls by accident".
Actually it's the `EmbeddedFormat.from` raising an `IllegalArgumentException`. I started creating a specific exception for that but then I noticed that another test was failing because an `IllegalArgumentException` raised by Kafka stuff (I have to double check which one) so I decided to handle them together.

Reply:
My point is that IAE can be thrown by anything. Because it's not a checked exception it's not really part of the contract of the methods you're calling. So it's possible that something throws IAE which should not be mapped to UNPROCESSABLE_ENTITY, but to something else.

Reply:
Yeah it's a good point. I added a specific exception. The test raising `IllegalArgumentException`, I mentioned, was raised by the wrong proto stuff where you did a comment. I reverted it to 500 even because we agreed that 422 is not ok.

Reply:
Does that mean you want a re-review?

Reply:
yes :-)